File: library/src/main/java/com/getstream/sdk/chat/logger/StreamLoggerLevel.java
package com.getstream.sdk.chat.logger;

public enum StreamLoggerLevel {

    /**
     * Show all Logs.
     */
    ALL(0),

    /**
     * Show DEBUG, WARNING, ERROR logs
     */
    DEBUG(1),

    /**
     * Show WARNING and ERROR logs
     */
    WARN(2),

    /**
     * Show ERROR-s only
     */
    ERROR(3),

    /**
     * Don't show any Logs.
     */
    NOTHING(4);

    private int severity;

    StreamLoggerLevel(int severity) {
        this.severity = severity;
    }

    public boolean isMoreOrEqualsThan(StreamLoggerLevel level) {
        return level.severity >= this.severity;
    }
}

File: app/src/main/java/io/getstream/chat/android/core/poc/library/logger/StreamLoggerLevel.kt
package io.getstream.chat.android.core.poc.library.logger

enum class StreamLoggerLevel(private val severity: Int) {
    /**
     * Show all Logs.
     */
    ALL(0),
    /**
     * Show DEBUG, WARNING, ERROR logs
     */
    DEBUG(1),
    /**
     * Show WARNING and ERROR logs
     */
    WARN(2),
    /**
     * Show ERROR-s only
     */
    ERROR(3),
    /**
     * Don't show any Logs.
     */
    NOTHING(4);

    fun isMoreOrEqualsThan(level: StreamLoggerLevel): Boolean {
        return level.severity >= severity
    }

}